ui/components/app/confirm/info/row/expandable-row.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+} from '../../../../component-library';
+import {
+  Display,
+  IconColor,
+} from '../../../../../helpers/constants/design-system';
+import { ConfirmInfoRow, ConfirmInfoRowProps } from './row';
+
+export type ConfirmInfoExpandableRowProps = ConfirmInfoRowProps & {
+  content: React.ReactNode | string;
+  startExpanded?: boolean;
+};
+
+export const ConfirmInfoExpandableRow = (
+  props: ConfirmInfoExpandableRowProps,
+) => {
+  const { content, children, startExpanded, ...rowProps } = props;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const ref = useRef<any>();
+
+  const [expanded, setExpanded] = useState<boolean>(Boolean(startExpanded));
+  const [, setLoaded] = useState<boolean>(false);
+
+  const handleClick = useCallback(() => {
+    setExpanded(!expanded);
+  }, [expanded]);
+
+  // Required to force a re-render so the content height can be calculated.
+  useEffect(() => {
+    setLoaded(true);
+  }, []);
```

Comment:
Can we add something to dependency array here, perhaps `expanded`?

Reply:
This is intentionally empty so it runs only once to calculate the height for the first render if it starts expanded.